### DISCUSSION TITLE
How to write the "target layer" for model densenet121?

### DISCUSSION THREAD

**Discussion by DONGTING98** [frgfm/torch-cam#200] [Q&A] [answered]
Hi!
I am trying to use SmoothGradCAMpp and I am confused at how to write the "target_layer". I found in most examples using resnet18, it could be "layer4". I tried "denseblock4" in my demo and it turned out to have error " ValueError: Unable to find all submodules ['denseblock4'] in the model". Can anyone help me? Thanks in advance.

Part of the structure of the densenet is as below, which might help. "Denseblock4" is the last block containing conv and it consists several sublayers named "denselayer1", "denselayer2" and so on. 
![image](https://user-images.githubusercontent.com/110180433/209541015-9325ab6d-a130-43ed-a349-3b2d59087a7e.png)

**Comment by DONGTING98** [comment, marked as answer]
I tried " features.denseblock4.denselayer16" and it turned out to work!

**Reply by frgfm** [reply]
Correct :+1: 
If passed as a string, that needs to be the period-separated module attributes of the top-level model !

Just in case, you can also pass the module as a reference directly:
```python
from torchcam.methods import LayerCAM
from torchvision.models import densenet121

model = densenet121(pretrained=True).eval()
extractor = LayerCAM(model, model.features.denseblock4.denselayer16)
```

**Reply by DONGTING98** [reply]
Thank you for replying!!
